src/Sarif.UnitTests/Visitors/MakeUriAbsoluteVisitorTest.cs

```diff
@@ -103,79 +103,117 @@ public void MakeUriAbsoluteVisitor_VisitPhysicalLocation_DoesNotSetUriIfBaseIsNo
+                ["%TEST2%"] = new FileLocation { Uri = new Uri(@"D:\bld\out\") }
             });
-            
+
             MakeUrisAbsoluteVisitor visitor = new MakeUrisAbsoluteVisitor();
 
             var newRun = visitor.VisitRun(run);
+
             // Validate.
-            newRun.Files[0].FileLocation.Uri.ToString().Should().Be(@"file://C:/srcroot/src/file1.cs");
-            newRun.Files[1].FileLocation.Uri.ToString().Should().Be(@"file://D:/bld/out/src/file2.dll");
-            newRun.Files[2].FileLocation.Uri.ToString().Should().Be(@"file://C:/srcroot/src/archive.zip");
-            newRun.Files[3].FileLocation.Uri.ToString().Should().Be(@"file://C:/srcroot/src/archive.zip#file3.cs");
-            newRun.Files[4].FileLocation.Uri.ToString().Should().Be(@"file://C:/srcroot/src/archive.zip#archive2.gz");
-            newRun.Files[5].FileLocation.Uri.ToString().Should().Be(@"file://C:/srcroot/src/archive.zip#archive2.gz/file4.cs");
-            newRun.Files[6].FileLocation.Uri.ToString().Should().Be(@"file://C:/srcroot/src/archive.zip#file5.cs");
+            newRun.Files[0].FileLocation.Uri.ToString().Should().Be(@"file:///C:/srcroot/src/file1.cs");
+            newRun.Files[1].FileLocation.Uri.ToString().Should().Be(@"file:///D:/bld/out/src/file2.dll");
+            newRun.Files[2].FileLocation.Uri.ToString().Should().Be(@"file:///C:/srcroot/src/archive.zip");
+            newRun.Files[3].FileLocation.Uri.ToString().Should().Be(@"file3.cs");
+            newRun.Files[4].FileLocation.Uri.ToString().Should().Be(@"archive2.gz");
+            newRun.Files[5].FileLocation.Uri.ToString().Should().Be(@"file4.cs");
 
             // Operation should zap all uri base ids
-            newRun.Files.Select(f => f.FileLocation.UriBaseId != null).Any().Should().BeFalse();
+            newRun.Files.Where(f => f.FileLocation.UriBaseId != null).Any().Should().BeFalse();
         }
 
         [Fact]
         public void MakeUriAbsoluteVisitor_VisitRun_DoesNotSetAbsoluteUriIfNotApplicable()
         {
-            Dictionary<string, Uri> uriMapping = new Dictionary<string, Uri>()
+            Dictionary<string, FileLocation> uriMapping = new Dictionary<string, FileLocation>()
             {
-                { "%TEST3%", new Uri(@"C:\srcroot\") },
-                { "%TEST4%", new Uri(@"D:\bld\out\") },
+                ["%TEST3%"] = new FileLocation { Uri = new Uri(@"C:\srcroot\") },
+                ["%TEST4%"] = new FileLocation { Uri = new Uri(@"D:\bld\out\") }
             };
 
             Run expectedRun = GenerateRunForTest(uriMapping);
             Run actualRun = expectedRun.DeepClone();
 
-
             MakeUrisAbsoluteVisitor visitor = new MakeUrisAbsoluteVisitor();
             var newRun = visitor.VisitRun(actualRun);
 
-            expectedRun.Should().Be(actualRun);
-        }
-
-        [Fact]
-        public void MakeUriAbsoluteVisitor_VisitRun_ThrowsIfPropertiesAreWrong()
```

Comment:
>MakeUriAbsoluteVisitor_VisitRun_ThrowsIfPropertiesAreWrong [](start = 20, length = 58)

no longer necessary. used to enforce that we formed the speculative originalUriBaseId dictionary properly. OM now enforces this, now that run.originalUriBaseIds is a thin.

Reply:
Got it.

---
In reply to: [247324768](https://github.com/Microsoft/sarif-sdk/pull/1210#discussion_r247324768) [](ancestors = 247324768)